src/components/Table/DataTable.tsx

```diff
@@ -1,23 +1,77 @@
 "use client";
 
-import { ColumnDef, flexRender, getCoreRowModel, useReactTable } from "@tanstack/react-table";
-import React, { useState } from "react";
+import React from "react";
+
+import {
+	ColumnDef,
+	CoreOptions,
+	ExpandedState,
+	flexRender,
+	getCoreRowModel,
+	getExpandedRowModel,
+	OnChangeFn,
+	RowSelectionState,
+	SortingState,
+	useReactTable
+} from "@tanstack/react-table";
+
 import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from "./Table";
+import { SortableHeader } from "./Sorting";
+import { cn } from "../..";
 
 interface DataTableProps<TData, TValue> {
 	columns: ColumnDef<TData, TValue>[];
 	data: TData[];
+	filters?: Record<string, string>;
+	resetFilters?: () => void;
+	getRowId?: CoreOptions<TData>["getRowId"];
+	rowSelection?: RowSelectionState;
+	onRowSelectionChange?: OnChangeFn<RowSelectionState>;
+	sorting?: SortingState;
+	onSortingChange?: OnChangeFn<SortingState>;
+	expanded?: ExpandedState;
+	onExpandedChange?: OnChangeFn<ExpandedState>;
+	getSubRows?: (row: TData) => TData[];
 }
 
-export function DataTable<TData, TValue>({ columns, data }: DataTableProps<TData, TValue>) {
-	const [rowSelection, setRowSelection] = useState({});
+export function DataTable<TData, TValue>({
+	columns,
+	data,
+	filters,
+	resetFilters,
+	sorting,
+	onSortingChange,
+	expanded,
+	onExpandedChange,
```

Comment:
You basically expose all the possible options here right?

Reply:
yep. this comes directly from pro